Update documentation to reflect Apama plugin changes (#1489)
thin-edge/thin-edge.io: docs/src/howto-guides/017_apama_software_management_plugin.md

# Apama Software Management Plugin

The apama plugin can be used to install and manage apama artefacts.
This plugin can be used to install an apama project on a device as well as inject monitor files into an already installed project.
The Apama plugin can be used to install Apama projects using the Cumulocity software management feature.

> Note: This plugin expects an Apama installation on the device.

## Install Apama artifacts from Cumulocity

> ## Download and Install Apama Plugin

Reviewer: Why did you remove this section on installing the Apama software management plugin itself? Is the plugin now part of the Apama installation itself? Otherwise, how to install it?

Author: The method of installation will change and I do not know what to put here yet. Some details have yet to be worked out, hence this is still a draft pull request.

Author: 91ba8fb

There's some naming convention that you need to follow while creating software entries for apama artefacts in the software repository.

For apama projects:

1. Name must be suffixed with `::project` as in `my-demo-project::project`
2. Version must be suffixed with `::apama` as in `1.0::apama` or just `::apama` if you don't need a version number.
3. The uploaded binary must be a `zip` file that contains the `project` directory. If a directory named `project` is not found at the root level in the zip, it's considered invalid.
1. The name must be suffixed with `::project` as in `my-demo-project::project`.
2. The version must be suffixed with `::Apama` as in `1.0::Apama` or just `::Apama` if no  version number is necessary.
3. The uploaded binary must be a `zip` file that contains the `project` directory. If a directory named `project` is not found by the plugin at the root level in the zip, it is considered invalid.

![Add new apama project in Software Repository](./images/apama-plugin/apama-project-c8y-software-repository.png)

> For apama monitor file:

Reviewer: Why was this section on injecting `mon` files removed? Doesn't the new version of the plugin support this anymore?

Author: Yes, the new version will not support this.

> /opt/softwareag/Apama/bin/Apama_env engine_inspect -m

Reviewer: Accidental side effect of "replace all"? Or has the binary name really changed?

# Apama Software Management Plugin

The Apama plugin, part of the Apama thin-edge support package, can be used to install Apama projects using the Cumulocity software management feature.

> ## Add the Apama thin-edge support package to the tenant software repository and deploy it to thin-edge

Reviewer: ```suggestion
## Prerequisites

The Apama thin-edge support package must be installed before you can install Apama projects on thin-edge.
If not already installed, it can also be installed from the Cumulocity software repository as follows:
```

> ## Add the Apama thin-edge support package to the tenant software repository and deploy it to thin-edge

Reviewer: You might also wanna provide a link to where the user can download this apama thin-edge support package. I also hope that this support package will work even on a machine where apama is already installed, in which case, it will just install the Apama thin-edge plugin only.

Reviewer 2: There's not a fixed URL for the package - it depends on the version. The install instructions should be to install the apama-repo package (for which there is a fixed URL) and then install it as here using apt from the repo

Author: c622e75

    - **Software**: apama-thin-edge-support
    - **Description**: apama-thin-edge-support (or something else if you choose)
    - **Device Filter Type**: (leave empty)
    - **Software Type**: apt
    - **Version**: 10.15.0.2-1::apt
    - **SOFTWARE FILE**: select the **Provide a file path** option and give an **empty space** (' ').
4. Click the **Add Software** button.
5. Now return to the **Devices** menu option and then select **All devices**.
6. In the list of devices, select the thin-edge device installed previously.
7. In the sub-menu for the device, select the **Software** option.
8. Click the **Install software** button in the bottom left; the apama-thin-edge-support should be listed.
9. Click the drop-down arrow on the right and check the 10.15.1.0-1::apama radio button, then click **Install**.
10. Finally, click the **Apply changes** button in the lower right of the panel.

After installation of the support for Apama onto the thin-edge device, the correlator will attempt to start, but will fail initially because there is not yet an Apama project installed. Instructions are given below for installing an Apama project on the device.

> ## Install Apama artifacts from Cumulocity

Reviewer: ```suggestion
## Install Apama projects from Cumulocity
```

Before an Apama project can be installed on the device using the software management feature in Cumulocity, the project files need to be added to the Cumulocity software repository.

There is a naming convention that you need to follow while creating software entries for Apama artifacts in the software repository.

For Apama projects:

> 1. The version must be suffixed with `::Apama` as in `1.0::Apama` or just `::Apama` if no  version number is necessary.

Reviewer: The earlier version of the `tedge-apama-plugin` that thin-edge team developed was installed at `/etc/tedge/sm-plugins` as `apama` instead of `Apama`. So, if you'd like to replace the existing installations of the old plugin with the new one, you may also wanna name your plugin binary as `apama` and that same name needs to be used as the suffix here. Otherwise both will coexist on machines where the older plugin is already installed. But, if the intent was to keep both side-by-side, then this is fine.

Reviewer 2: Pretty sure we are using lowercase and this is just a doc issue. The package will force-remove the tedge-apama-plugin when it installs

Author: The file is lower case. This is a docs error caused by VS code, which I will fix.

Author: a8d5ca5

# Apama Software Management Plugin

The Apama plugin, part of the Apama thin-edge support package, can be used to install Apama projects using the Cumulocity software management feature.

## Add the Apama thin-edge support package to the tenant software repository and deploy it to thin-edge

1. In the **Device Management** app and go to the **Management** menu option and select the **Software repository**.
2. Click **Add software** at the right of the top menu bar.
3. In the **ADD SOFTWARE** dialog enter the following details:
    - **Software**: apama-thin-edge-support
    - **Description**: apama-thin-edge-support (or something else if you choose)
    - **Device Filter Type**: (leave empty)
    - **Software Type**: apt

>     - **Version**: 10.15.1.0-1::apt

Reviewer 2: This is the version from the 10.15.1 branch - it should be 10.15.0.2 in the published version

Author: cc887df

5. Now return to the **Devices** menu option and then select **All devices**.
6. In the list of devices, select the thin-edge device installed previously.
7. In the sub-menu for the device, select the **Software** option.
8. Click the **Install software** button in the bottom left; the apama-thin-edge-support should be listed.
9. Click the drop-down arrow on the right and check the 10.15.1.0-1::apama radio button, then click **Install**.
10. Finally, click the **Apply changes** button in the lower right of the panel.

After installation of the support for Apama onto the thin-edge device, the correlator will attempt to start, but will fail initially because there is not yet an Apama project installed. Instructions are given below for installing an Apama project on the device.

## Install Apama artifacts from Cumulocity

Before an Apama project can be installed on the device using the software management feature in Cumulocity, the project files need to be added to the Cumulocity software repository.

There is a naming convention that you need to follow while creating software entries for Apama artifacts in the software repository.

For Apama projects:

> 1. The name must be suffixed with `::project` as in `my-demo-project::project`.

Reviewer: This `::project` suffix and `::mon` suffix were needed only when the plugin was supporting the installation of both projects and mon files. But, now that the plugin only supports project files **only**, you can avoid this suffix requirement.

Author: cb58271

# Apama Software Management Plugin

The Apama plugin, part of the Apama thin-edge support package, can be used to install Apama projects using the Cumulocity software management feature.

## Prerequisites

The Apama thin-edge support package must be installed before you can install Apama projects on thin-edge.

> The recommended way of installing Apama support is with the Debian based distributions apt package manager.

Reviewer: ```suggestion
The recommended way of installing Apama support on Debian based distributions is with the apt package manager.
```

Further details about this can be found in the thin-edge.io documentation at [Manage the software on your devices from Cumulocity cloud](https://thin-edge.github.io/thin-edge.io/html/tutorials/software-management.html).

> #### Set up the repository from which the packages will be installed

Reviewer: ```suggestion
#### Set up the Debian repository from which the Apama packages will be installed

```

1. In the Cumulocity IoT tenant, open the **Device Management** app, go to the **Management** menu option and select the **Software repository**.
2. Click **Add software** at the right of the top menu bar. 
3. In the **ADD SOFTWARE** dialog enter the following details:
- **Software**: apama-repo
- **Description**: apama-repo (or something else if you choose)
- **Device Filter Type**: (leave empty)
- **Software Type**: apt
- **Version**: 2022::apt

> - **SOFTWARE FILE**: check the Provide a file path option and enter the URL:	https://downloads.apamacommunity.com/debian/apama-repo_2022_all.deb

Reviewer: ```suggestion
- **SOFTWARE FILE**: check the *Provide a file path* option and enter the URL:	https://downloads.apamacommunity.com/debian/apama-repo_2022_all.deb
```
